[Feat/myp 01] 마이페이지 ui 구현

heehee-ju opened the pull request:
## 📌 관련 이슈 (Task ID)

- ID: Feat/myp 01
## 📝 작업 내용 (Details)

- [x] 활동 통계 컴포넌트(ActivityStats) 구현 - 캡슐, 이스터에그, 친구 통계 표시
- [x] 로그아웃 버튼 컴포넌트(LogoutButton) 구현 - 공통 Button 컴포넌트 활용
- [x] 메뉴 리스트 컴포넌트(MenuList) 구현 - 설정, 결제 내역, 고객 센터 메뉴
- [x] 프로필 섹션 컴포넌트(ProfileSection) 업데이트 - 프로필 이미지, 닉네임, 이메일 표시
- [x] 마이페이지 메인 컨테이너(MyPageFeature) 구조 개선 및 컴포넌트 통합
- [x] 각 컴포넌트별 스타일 파일(styles.ts) 추가 및 스타일링 완료
- [x] Feature Slice Architecture 패턴 준수 - 컴포넌트 분리 및 구조화

## 📸 스크린샷 (Screenshots)

(스크린샷 추가 예정)

## 🧐 리뷰 포인트 (Review Point)

- 마이페이지 컨테이너에서 로그아웃 핸들러가 TODO 상태입니다. 실제 로그아웃 로직 구현이 필요합니다.
- 활동 통계의 데이터는 현재 하드코딩되어 있습니다. 추후 API 연동이 필요합니다.
- 프로필 섹션의 사용자 정보(닉네임, 이메일)도 하드코딩되어 있습니다. 실제 사용자 데이터 연동이 필요합니다.
- 메뉴 리스트의 각 메뉴 항목 클릭 이벤트 처리가 아직 구현되지 않았습니다.

## ✅ 체크리스트 (Checklist)

- [x] 컨벤션(커밋, 브랜치 네이밍)을 준수했나요?
- [x] 불필요한 콘솔 로그(console.log)나 주석을 제거했나요?
- [ ] 실행 시 에러가 발생하지 않나요?

heehee-ju committed:
chore(mypage): 마이페이지 폴더 구조 생성 및 라우팅 연결

- components/mypage 폴더 구조 생성
  - components: profile-section, activity-stats, menu-list, logout-button
  - hooks: useUserInfo, useLogout
  - types.ts, index.tsx
- app/(tabs)/mypage.tsx 라우팅 수정
  - @/components/mypage에서 MyPageFeature import

heehee-ju committed:
feat: 마이페이지 프로필 섹션 컴포넌트 구현

- MyPageFeature 컨테이너 컴포넌트 추가
- ProfileSection 컴포넌트 구현 (Figma 디자인 1:1 대응)
- 프로필 이미지, 사용자명, 이메일 표시
- React Native StyleSheet 기반 스타일링
- 색상/타이포그래피 토큰 활용
- flexbox 레이아웃 적용

heehee-ju committed:
fix: MyPage에 SafeAreaView 적용 및 RootProvider 구조 수정

- MyPage 컴포넌트에 SafeAreaView 적용하여 상단 safe area 처리
- RootProvider에서 SafeAreaView를 SafeAreaProvider 밖에 두었던 구조 수정
- 불필요한 prompt 파일 삭제 및 profile-section prompt 추가

heehee-ju committed:
feat: 버튼 컴포넌트, 맵 기능 개선, 결제 시스템 리팩토링 및 타임캡슐 생성 기능 업데이트

heehee-ju committed:
feat: 마이페이지 컴포넌트 구현 (활동 통계, 로그아웃 버튼, 메뉴 리스트, 프로필 섹션)

Daram0o0 commented on components/mypage/components/profile-section/styles.ts:
```diff
@@ -0,0 +1,132 @@
+    color: Colors.black[500],
+    fontFamily: 'Pretendard Variable',
+    fontWeight: '400',
+    letterSpacing: 0.263671875,
```
폰트도 최대한 constants내부에 있는 폰트 위주로 사용해주세용!

Daram0o0 reviewed (commented):
전체적으로 파일 잘 나눠져 있는데 폰트 부분만 constants 내부 typography.ts쓰는 걸로 변경하면 조금 더 좋을 것 같슴다!

heehee-ju committed:
feat: 로그아웃 버튼 기능 구현 및 아키텍처 규칙 준수

heehee-ju committed:
design: 활동 통계 컴포넌트 피그마 디자인 반영 및 프롬프트 수정

- activity-stats 프롬프트 파일을 React Native 프로젝트 구조에 맞게 수정
- 피그마 MCP 연동하여 activity-stats 컴포넌트 디자인 수정
  - 컨테이너 padding, gap 값 피그마 디자인에 맞게 조정
  - 통계 값 폰트 weight, lineHeight, letterSpacing 피그마 값 반영
  - 색상 토큰 사용 (Colors, Typography, Spacing, FontWeight)
- 관련 컴포넌트 스타일 일관성 개선
